### DISCUSSION TITLE
Change Quarto chunk options from within an R function?

### DISCUSSION THREAD

**Discussion by maxdrohde** [quarto-dev/quarto-cli#964] [Q&A] [answered]
Anyone have an idea if it's possible to set a Quarto chunk option from within an R function?

For a minimal example, I have an R function that returns raw markdown output for a header.

````
```{r}
make_header <- function(text) cat(paste("##", text))
```
````
To get this to work, you must set `#| output: asis` in the code cell to render as markdown output.
````
```{r}
#| output: asis

make_header("my_header_text")
```
````

Is there a way to set `#| output: asis` within the `make_header()` function so that the chunk option doesn't need to be set manually each time?

**Comment by jjallaire** [comment, marked as answer]
Knitr has an asis_output function you can use: https://www.rdocumentation.org/packages/knitr/versions/1.39/topics/asis_output

**Reply by cderv** [reply]
To add on this answer, you'll find examples of this function used to produce raw content in the R Markdown cookbook:

* https://bookdown.org/yihui/rmarkdown-cookbook/latex-html.html
* https://bookdown.org/yihui/rmarkdown-cookbook/opts-render.html
